testsuite/mlir/mlir-parse/dialect.mlir

```diff
@@ -23,38 +23,40 @@
 // }
 // ```
 
+!C = type !verona.class<"C">
+!D = type !verona.class<"D", "f" : meet<U64, imm>, "g" : meet<!C, mut>>
```

Comment:
If the `!verona.class<>` completely defines the class, why do we need the dialect `verona.class { }` construct with the same info? Seems to me that if we don't keep them in sync, weird stuff is going to happen.

Reply:
The allocate operation is what will tie them together. You can only allocate an object by giving it an implementation that has the right signature.

Reply:
That's what I don't like. If we change the implementation of the class (say extract fields to create lists), we'll have to change the operations and the types in the exact same way. It's twice the amount of work for no additional benefit.